pages/bookmarks.jsx

```diff
@@ -3,54 +3,50 @@ import { getBookmarks } from '../js/bookmarks-manager';
 import ProjectList from '../components/project-list/project-list.jsx';
 import HintMessage from '../components/hint-message/hint-message.jsx';
 
-import UserData from '../js/user-data.js';
-import Login from '../js/login.js';
+import user from '../js/app-user.js';
 import env from "../config/env.generated.json";
 
 export default React.createClass({
   getInitialState() {
     return {
-      bookmarks: null
+      bookmarks: null,
+      user
     };
   },
+  handleSignInBtnClick(event) {
+    event.preventDefault();
+
+    let redirectUrl = `${env.ORIGIN}${this.props.router.getCurrentLocation().pathname}`;
```

Comment:
Hmm, doesn't this code only work in a browser, when a user clicks a button? It kind of feels like `user.login()` should know to use `window.location.toString()` as its redirect URL, unless there is no `window` variable available... thoughts?